### PR TITLE
Tidying custom hook

### PR DESCRIPTION
**Pull request by valtat**
In this pull request I have attempted to separate the app logic into two separate custom hooks - useCurrencyConverter for calculating the exchange rate and useFetch to fetch the currency data from exchange rates api. This has been done in order to increase the DX and cleanliness of the code, however the results of said changes is yet unclear. Hopefully the code is now more modular and the reusability of the components is greater, and the readability is improved upon. I make no promises though! The commits might be a bit confusing, since I had big issues making everything work, so I had to refactor the code time and time again to make everything work. The earlier commits might reflect that.

### CODE REVIEW

**Comment by baldez300** [comment]
The commit messages provide a clear and concise overview of the changes made to the codebase. Each commit appears to have a specific purpose, and the comments within the commits enhance understanding. 

The restructuring of the code into different files and folders, such as moving the useCurrencyConverter custom hook to the "**hooks**" folder, contributes to a more organized project structure and demonstrates an awareness of code cleanliness.

The overall code structure adheres to best practices, and I appreciates the use of React hooks for managing state and side effects. Good job..!

However, it's important to note that in the commit with the message "**Removed unnecessary imports/lines,**"  what motivated their removal?

Looking at the code in `App.js`, how does the application handle unexpected inputs or exceptions, particularly in the context of the custom hooks (`useCurrencyConverter `and `useFetch`)?

**Comment by TaiNguyen2407** [comment]
I think the PR gives quite clear what its goals are and what you are trying to achieve. 

Commit messages are clear, they follow a certain standard. 

Reorganization of the codebase is well-thought and adheres to great guidelines. 

Overall I think the PR follows DX guidelines, even though there is confusion due to refactoring but you mentioned it in PR description so I think it's great. 

I also added a few questions to the code changes, feel free to have a look

Great work!

File: src/App.js [hunk 1]
```diff
@@ -1,28 +1,43 @@
...
+};
+
+const BASE_URL =
+  "http://api.exchangeratesapi.io/v1/" +
+  params.endpoint +
+  "?access_key=" +
+  params.access_key;
 
 function App() {
+  const [amount, setAmount] = useState(1);
```
Comment (TaiNguyen2407) [inline src/App.js:21]: what is the purpose of this amount variable here and why is it set to 1 in the first place

File: src/hooks/useCurrencyConverter.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { useState, useEffect } from "react";
+
+function calculateExchangeRate(currencies, fromCurrency, toCurrency) {
+  if (!currencies || !currencies.rates) {
+    return;
+  }
+  if (fromCurrency === currencies.base) {
```
Comment (TaiNguyen2407) [inline src/hooks/useCurrencyConverter.js:7]: Can you explain more about your choice of having a series if else here ? Can we use case or other methods somehow ?

File: src/App.js (hunk 1 above)
Comment (valtat) [reply src/App.js:21]: The purpose of this variable is the amount displayed in the upper input field, which is given the parameter of 1 in order to display a value of 1 as the initial number

File: src/hooks/useCurrencyConverter.js (hunk 2 above)
Comment (valtat) [reply src/hooks/useCurrencyConverter.js:7]: The checking is a bit messy here, which I admit. The purpose is to first check if the currency object is defined (and the rates inside) and return if it's undefined, then check if the fromCurrency is the same as the base currency (which is EUR in the api, btw) and if so, return the exchange rate for the toCurrency from the object, and then check if toCurrency is equal to the base currency, in which case it returns the inverse of the exchange rate for fromCurrency. Finally if neither fromCurrency or toCurrency is the base currency, the function will calculate the exchange rate between the two currencies by dividing their exchange rates.

I'm all about new ideas to simplify this conversion, might be worth it to explore a switch case or another way to make it more graceful

**Pull request merged**